modules/space/kepler.cpp

```diff
@@ -403,6 +404,108 @@ namespace {
         return
             nSecondsSince2000 + totalSeconds + nLeapSecondsOffset - offset + date.seconds;
     }
+
+    std::string unpackDate(std::string_view packedDate) {
+        // Some data in the MPC dataset are stored in a packed data format that this
+        // function unpacks. More information on packed dates can be found here:
+        // http://www.minorplanetcenter.org/iau/info/PackedDates.html
+
+        if (packedDate.size() < 5) {
+            throw ghoul::RuntimeError(std::format(
+                "Illformed packed date. Size must be 5 characters. {}", packedDate
+            ));
+        }
+
+        int year = [packedDate](char m) {
```

Comment:
Nice! Should still probably multiply by 100 so that we get the correct year at line 438?